server/src/main/java/io/crate/metadata/TransactionContext.java

```diff
@@ -23,38 +23,31 @@
 package io.crate.metadata;
 
 import io.crate.metadata.settings.SessionSettings;
-import org.joda.time.DateTimeUtils;
 
-public interface TransactionContext {
+import java.time.Instant;
 
-    static TransactionContext of(SessionSettings sessionSettings) {
-        return new StaticTransactionContext(sessionSettings);
-    }
-
-    long currentTimeMillis();
+public class TransactionContext {
```

Comment:
Could we keep this as an itnerface?

Due to the change to caching the `Instant` I think the remaining changes to share more logic are no longer necessary.

That would avoid the `super(null);` in the `CoordinatorTxnCtx`  which seems like a hack?

Reply:
I agree, it looks like a hack... changing.